backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/RabbitMQHealthCheck.java

```diff
@@ -43,15 +45,13 @@ public ComponentName componentName() {
     }
 
     @Override
-    public Result check() {
-        try {
-            if (connectionPool.tryConnection() && rabbitChannelPoolImpl.tryChannel()) {
-                return Result.healthy(COMPONENT_NAME);
-            } else {
-                return Result.unhealthy(COMPONENT_NAME, "The created connection was not opened");
-            }
-        } catch (Exception e) {
-            return Result.unhealthy(COMPONENT_NAME, "Unhealthy RabbitMQ instances: could not establish a connection", e);
-        }
+    public Mono<Result> check() {
+        return Mono.fromCallable(() -> connectionPool.tryConnection() && rabbitChannelPoolImpl.tryChannel())
+            .filter(isOpened -> isOpened)
+            .flatMap(any -> Mono.just(Result.healthy(COMPONENT_NAME)))
```

Comment:
`then`?

Reply:
then() would always execute whether the filter return a value or not, so we can not switch in the case isOpened == false. I used then() at first but it caused some test to fail.

Reply:
why not `map` ?